packages/dart/test/sentry_logger_test.dart

```diff
@@ -183,22 +183,103 @@ void main() {
         'special values {"nan": NaN, "positive_infinity": Infinity, "negative_infinity": -Infinity}');
     expect(logCall.logger, 'sentry_logger');
   });
+
+  test('applies scope attributes to all log methods', () {
+    final logger = fixture.getSut();
+
+    final scopeAttributes = <String, SentryAttribute>{
+      'scopeString': SentryAttribute.string('fromScope'),
+      'scopeInt': SentryAttribute.int(42),
+    };
+    fixture.hub.scope.setAttributes(scopeAttributes);
+
+    logger.trace('trace message');
+    logger.debug('debug message');
+    logger.info('info message');
+    logger.warn('warn message');
+    logger.error('error message');
+    logger.fatal('fatal message');
+
+    expect(fixture.hub.captureLogCalls.length, 6);
+    for (final call in fixture.hub.captureLogCalls) {
+      final attrs = call.log.attributes;
+      expect(attrs['scopeString'], scopeAttributes['scopeString']);
+      expect(attrs['scopeInt'], scopeAttributes['scopeInt']);
+    }
+  });
+
+  test('per-log attributes override scope on same key', () {
+    final logger = fixture.getSut();
+
+    final scopeAttributes = <String, SentryAttribute>{
+      'overridden': SentryAttribute.string('fromScope'),
+      'kept': SentryAttribute.bool(true),
+    };
+    fixture.hub.scope.setAttributes(scopeAttributes);
+
+    final overrideAttr = SentryAttribute.string('fromLog');
+    logger.info('override test', attributes: {
+      'overridden': overrideAttr,
+      'logOnly': SentryAttribute.double(1.23),
+    });
+
+    expect(fixture.hub.captureLogCalls.length, 1);
+    final captured = fixture.hub.captureLogCalls[0].log;
+    // overridden key should come from per-log attributes
+    expect(captured.attributes['overridden'], overrideAttr);
+    // scope-only key should still be present
+    expect(captured.attributes['kept'], scopeAttributes['kept']);
+    // log-only key should be present
+    expect(captured.attributes['logOnly']?.type, 'double');
+  });
+
+  test('formatter path merges template, per-log and scope attributes', () {
+    final logger = fixture.getSut();
+
+    final scopeAttributes = <String, SentryAttribute>{
+      'scopeOnly': SentryAttribute.string('present'),
+    };
+    fixture.hub.scope.setAttributes(scopeAttributes);
+
+    logger.fmt.info(
```

Comment:
I guess we could also add a test for the case that `logger.ftm` sets the same attribute as `logger`. I guess last one wins here?